src/ZODB/DB.py

```diff
@@ -1010,8 +1010,13 @@ def commit(self, transaction):
                 self._oids.update(result[1])
 
     def tpc_vote(self, transaction):
-        for oid, _ in  self._storage.tpc_vote(transaction) or ():
-            self._oids.add(oid)
+        result = self._storage.tpc_vote(transaction)
+        if result:
+            if isinstance(result[0], bytes):
+                self._oids.update(set(result))
```

Comment:
I didn't think so, but I failed with this the other day. (The think I know now what my faulty flail was.) Fixed, and also below.